.github/workflows/test.yml

```diff
@@ -16,20 +16,23 @@ jobs:
   test:
     name: "Run unit tests for reference code"
     runs-on: ubuntu-latest
-    container:
-      image: sagemath/sagemath:latest
-      options: --user root
     steps:
     - name: Checkout
       uses: actions/checkout@v3
 
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+        cache: pip
+
     - name: Install dependencies
-      run: sage --pip install pycryptodomex mypy
+      run: python -m pip install pycryptodomex mypy matplotlib
```

Comment:
Why is matplotlib needed?
If this is strictly for `plot_prio3_multiproof_robustness.py`, then IMO we're better reomving this script rather than adding the dependency.